cmake/list_append_keys_from.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
+#
+macro(list_append_keys_from listname source)
```

Comment:
The calling code will be almost identical. Either you call `list_append_keys_from(targte_list_name source_list_name)` or `list_append_keys_from(targte_list_name ${source_list_name})`. But the second signature also support calling it without having a source list in the first place but just plain items, e.g. `list_append_keys_from(targte_list_name "one" "two" "three" ${or_even_another_list})`.

The implemenation is very similar to [list_append_unique](https://github.com/ros/catkin/blob/184317881a0b6eb28d4386b791a2375c02da7d2a/cmake/list_append_unique.cmake#L9-L10), simply loop over `${ARGN}`.